File: python/space-age/space_age.py
class SpaceAge:
    def __init__(self, seconds):
        self.seconds = seconds

    def on_earth(self):
        period_days = 365.25
        return self.calculate_years(period_days)

    def on_mercury(self):
        period_days = 0.2408467 * 365.25
        return self.calculate_years(period_days)

    def on_venus(self):
        period_days = 0.61519726 * 365.25
        return self.calculate_years(period_days)

    def on_mars(self):
        period_days = 1.8808158 * 365.25
        return self.calculate_years(period_days)

    def on_jupiter(self):
        period_days = 11.862615 * 365.25
        return self.calculate_years(period_days)

    def on_saturn(self):
        period_days = 29.447498 * 365.25
        return self.calculate_years(period_days)

    def on_uranus(self):
        period_days = 84.016846 * 365.25
        return self.calculate_years(period_days)

    def on_neptune(self):
        period_days = 164.79132 * 365.25
        return self.calculate_years(period_days)

    def calculate_years(self, period):
        return round(self.seconds / (period*24*60*60), 2)

File: java/space-age/src/main/java/SpaceAge.java
class SpaceAge {

    double ageSeconds;

    public static final double EARTH_ORBITAL_PERIOD = 31557600;
    public static final double MERCURY_ORBITAL_PERIOD = EARTH_ORBITAL_PERIOD * 0.2408467;
    public static final double VENUS_ORBITAL_PERIOD = EARTH_ORBITAL_PERIOD * 0.61519726;
    public static final double MARS_ORBITAL_PERIOD = EARTH_ORBITAL_PERIOD * 1.8808158;
    public static final double JUPITER_ORBITAL_PERIOD = EARTH_ORBITAL_PERIOD * 11.862615;
    public static final double SATURN_ORBITAL_PERIOD = EARTH_ORBITAL_PERIOD * 29.447498;
    public static final double URANUS_ORBITAL_PERIOD = EARTH_ORBITAL_PERIOD * 84.016846;
    public static final double NEPTUNE_ORBITAL_PERIOD = EARTH_ORBITAL_PERIOD * 164.79132;

    SpaceAge(double seconds) {
        this.ageSeconds = seconds;
    }

    double getSeconds() {
        return this.ageSeconds;
    }

    double onEarth() {
        return this.ageSeconds / EARTH_ORBITAL_PERIOD;
    }

    double onMercury() {
        return this.ageSeconds / MERCURY_ORBITAL_PERIOD;
    }

    double onVenus() {
        return this.ageSeconds / VENUS_ORBITAL_PERIOD;
    }

    double onMars() {
        return this.ageSeconds / MARS_ORBITAL_PERIOD;
    }

    double onJupiter() {
        return this.ageSeconds / JUPITER_ORBITAL_PERIOD;
    }

    double onSaturn() {
        return this.ageSeconds / SATURN_ORBITAL_PERIOD;
    }

    double onUranus() {
        return this.ageSeconds / URANUS_ORBITAL_PERIOD;
    }

    double onNeptune() {
        return this.ageSeconds / NEPTUNE_ORBITAL_PERIOD;
    }

}
